src/Microsoft.Diagnostics.Tools.RuntimeClient/DiagnosticHelpers.cs

```diff
@@ -76,66 +86,65 @@ public static int AttachProfiler(int processId, uint attachTimeout, Guid profile
                 Pid = (uint)Process.GetCurrentProcess().Id,
             };
 
-            byte[] serializedConfiguration;
-            using (var stream = new MemoryStream())
+            byte[] serializedConfiguration = SerializeProfilerAttach(attachTimeout, profilerGuid, profilerPath, additionalData);
+            var message = new IpcMessage(DiagnosticServerCommandSet.Profiler, (byte)ProfilerCommandId.AttachProfiler, serializedConfiguration);
+
+            var response = IpcClient.SendMessage(processId, message);
+
+            var hr = 0;
+            switch ((DiagnosticServerCommandId)response.Header.CommandId)
             {
-                serializedConfiguration = SerializeProfilerAttach(header, stream, attachTimeout, profilerGuid, profilerPath, additionalData);
+                case DiagnosticServerCommandId.Error:
+                    // Handle error differently?
```

Comment:
What do you want to do differently here?
Let's either do it or log an issue to do it and left a TODO comment.